src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -297,18 +298,44 @@ public int getRepairSessionSpaceInMegabytes()
         return DatabaseDescriptor.getRepairSessionSpaceInMiB();
     }
 
+    @Deprecated
     @Override
     public void setRepairSessionSpaceInMebibytes(int sizeInMebibytes)
     {
         DatabaseDescriptor.setRepairSessionSpaceInMiB(sizeInMebibytes);
     }
 
+    @Deprecated
     @Override
     public int getRepairSessionSpaceInMebibytes()
     {
         return DatabaseDescriptor.getRepairSessionSpaceInMiB();
     }
 
+    @Override
+    public void setRepairSessionSpaceInMiB(int sizeInMebibytes)
+    {
+        try
+        {
+            DatabaseDescriptor.setRepairSessionSpaceInMiB(sizeInMebibytes);
+        }
+        catch (ConfigurationException e)
+        {
+            throw new IllegalArgumentException(e.getMessage());
```

Comment:
It's better to use `IllegalArgumentException(Throwable cause)` constructor, because it will preserve `e` stacktrace.

Reply:
This would be a precedent, I can't imagine putting the Stacktrace as an error message in JConsole, @dcapwell ?

Reply:
Unless I'm missing something, I don't think this will be applicable here.  I tried it and JConsole did not respond well to it and had trouble properly displaying the stacktrace as is.